lldb/source/Plugins/Process/elf-core/ProcessElfCore.cpp

```diff
@@ -1119,3 +1127,30 @@ bool ProcessElfCore::GetProcessInfo(ProcessInstanceInfo &info) {
   }
   return true;
 }
+
+std::optional<CoreNote> ProcessElfCore::GetAmdGpuNote() {
+  ObjectFileELF *core = (ObjectFileELF *)(m_core_module_sp->GetObjectFile());
+  if (core == nullptr)
+    return std::nullopt;
+  llvm::ArrayRef<elf::ELFProgramHeader> program_headers =
+      core->ProgramHeaders();
+  if (program_headers.size() == 0)
+    return std::nullopt;
+
+  for (const elf::ELFProgramHeader &H : program_headers) {
+    if (H.p_type == llvm::ELF::PT_NOTE) {
+      DataExtractor segment_data = core->GetSegmentData(H);
+      llvm::Expected<std::vector<CoreNote>> notes_or_error =
+          parseSegment(segment_data);
+      if (!notes_or_error) {
+        llvm::consumeError(notes_or_error.takeError());
+        continue;
+      }
+      for (const auto &note : *notes_or_error) {
+        if (note.info.n_type == llvm::ELF::NT_AMDGPU_KFD_CORE_STATE)
```

Comment:
You can't just check the `note.info.n_type`, you also need to match the name first. Every note type with a different name can re-use the same value for `note.info.n_type`.